### DISCUSSION TITLE
Weitere grafische Aufbereitung des Loggings

### DISCUSSION THREAD

**Discussion by wiggal** [wiggal/GEN24_Ladesteuerung#42] [Ideas]
Hier könnt ihr mir Ideen über neue grafische Aufbereitungen zum Logging,
bzw. Verbesserungen zu bestehenden Grafiken mitteilen.

Schönene Grüße 
Wiggal

**Comment by wiggal** [comment]
Hab mich nun mal an das leidige Thema  Abbildung der Energieflüsse bei Laden aus dem Netz bzw. Einspeisen aus dem Akku gemacht.

Mit der Darstellung wie im Solarweb nach Produktion und Verbrauch bin ich nicht weitergekommen.
Hier fehlen immer irgend welche Daten, die man nicht im Diagramm unterbringt.

Ich habe das Diagramm nun in EnergieQuellen (Darstellung negativ)  und EnergieZiele (Darstellung positiv) eingeteilt.

hier mal Screenshots vom Entwicklungsstand, vielleich hat jemand eine Meinung oder Tipp's dazu:
![grafik](https://github.com/wiggal/GEN24_Ladesteuerung/assets/69506553/87f94185-03f0-4338-8161-bc4e78b58023)
![grafik](https://github.com/wiggal/GEN24_Ladesteuerung/assets/69506553/aab8dbb0-58d9-414b-a8de-0f6fae175160)

**Comment by wiggal** [comment]
Hier der aktuelle Stand von Diagramm2 aus der Version 0.12.4

![grafik](https://github.com/wiggal/GEN24_Ladesteuerung/assets/69506553/6deedc02-cbb2-4d48-9103-2674acd694ae)

![grafik](https://github.com/wiggal/GEN24_Ladesteuerung/assets/69506553/c6e8e164-a452-4e6c-9a7d-3c4b01f361e2)

**Comment by iReckyy** [comment]
Ist es möglich und/oder in Planung die Daten des Ohm/Wattpiloten in der Grafik mit anzuzeigen?

**Reply by wiggal** [reply]
Hallo @iReckyy,

geplant habe ich diesbezüglich nichts, aber um sowas umzusetzen wäre als Erstes wichtig die Daten zu bekommen.
Ich hab in der API des GEN24 diesbezüglich noch nichts gesehen. Vielleicht kennst du ja die API, die die Daten liefert.

Schönen Gruß

**Reply by iReckyy** [reply]
Kannst du mit diesem Teil etwas anfangen?





{
				"attributes" : 
				{
					".simple-version-field" : "OHMP;;4c;",
					"[ENABLE]" : "true",
					"[VISIBLE]" : "true",
					"addr" : "255",
					"bootloader_sw_version" : "210",
					"co_processor_bootldr_version" : "1",
					"co_processor_sw_version" : "32",
					"connection" : "{\"id\":\"192.168.178.13:502\",\"ip\":\"192.168.178.13\",\"name\":\"modbus-tcp\",\"port\":502,\"protocol\":\"ModbusTCP\"}\n",
					"createTS" : "1719333665",
					"id" : "192.168.178.13:502@255",
					"if" : "modbus-tcp;192.168.178.13:502",
					"main" : "this",
					"main_processor_sw_version" : "76",
					"manufacturer" : "Fronius",
					"model" : "Ohmpilot",
					"serial" : "xxxxxxxxx",
					"this.rev-hw" : "6",
					"this.rev-sw" : "1.0.25-3",
					"wlan_modul_sw_version" : "190502"
				},
				"channels" : 
				{
					"COMPONENTS_MODE_ENABLE_U16" : 1.0,
					"COMPONENTS_MODE_VISIBLE_U16" : 1.0,
					"COMPONENTS_TIME_STAMP_U64" : 1719566225.0,
					"OHMPILOT_ENERGYACTIVE_CONSUMED_SUM_F64" : 9219301200.0,
					"OHMPILOT_MODE_CODE_OF_STATE_F64" : 0.0,
					"OHMPILOT_POWERACTIVE_DESIRED_F64" : 2525.5151068872251,
					"OHMPILOT_POWERACTIVE_SUM_MEAN_F32" : 2098.0,
					"OHMPILOT_TEMPERATURE_CHANNEL_01_MEAN_F64" : 0.0
				}
			}
		}
	},
	"Head" : 
	{
		"Note" : "this internal API may be changed any time",
		"RequestArguments" : 
		{
			"method" : "GET",
			"resource" : "readable"
		},
		"Status" : 
		{
			"Code" : 0,
			"ErrorDetail" : 
			{
				"Nodes" : []
			},
			"Reason" : "",
			"UserMessage" : ""
		},
		"Timestamp" : "2024-06-28T09:17:05+00:00"

**Reply by wiggal** [reply]
von welcher API, kommt das? Vom GEN24? 
`OHMPILOT_ENERGYACTIVE_CONSUMED_SUM_F64" : 9219301200.0,"`
Das müsste der Zählerstand sein, was der Ohmpilot bisher verbraucht hat, kannst du dir vorstellen was das für eine Einheit ist? Watt pro Sekunde ist das z.B. beim GEN24

**Reply by iReckyy** [reply]
Das kommt vom Gen24.

Habe selber noch nicht genauer geguckt.

**Reply by iReckyy** [reply]
Ja, könnte hinkommen, habe aus Solar.web mal die Verbräuche über die Jahre addiert und die kWh in kWs umgerechnet.
Komme dann auf ~9.180.000.
solar.web rundet stark.

Allerdings fehlen dann zwei Nullen.
Oder sind das Platzhalter?

Moment, Watt/Sekunde, nicht Kilowattsekunden….

**Reply by iReckyy** [reply]
Hallo, ist es inzwischen evtl. möglich den Ohmpiloten mit anzeigen zu lassen?
Wenn noch Daten dazu gebraucht werden unterstütze ich gerne.

**Reply by wiggal** [reply]
Hallo @iReckyy,
das geht leider nicht, weder Ohmpilot noch Wattpilot.

Ist halt schwierig, dazu müssten wir die Datenbank erweitern, und die API liefert die Werte nur, wenn es einen Ohmpiloten gibt.
Da müsste ich die Datenbank umbauen, und die API-Abrufe anpassen.
 Wiggal

**Reply by iReckyy** [reply]
Hm, schade.

**Comment by wiggal** [comment]
Hi, momentan mach ich nichts an den Diagrammen, wenn ich daran mal was stricke (evtl. Winter) meld ich mich, diesbezüglich wieder

**Reply by iReckyy** [reply]
Alles klar.

**Reply by roethigj** [reply]
Nun ist ja Winter...
Nur eine kleine Anfrage: Wäre es nicht sinnvoll die Prognose (der Produktion) auch negativ anzeigen zu lassen?

**Reply by wiggal** [reply]
Der Winter ist leider schon vorbei, aber Spaß beiseite.

Ich habe schon mal darüber nachgedacht, die Prognose **nur** negativ darzustellen, da sie ja eher was mit der Quelle als mit dem Ziel der Energie zu tun hat.

**Reply by roethigj** [reply]
> Prognose **nur** negativ

So meinte ich das. Das "Auch" bezog sich auf die PV-Produktion.
Ich hatte schon mal versucht die Werte für Vorhersage in den SQL-Abfragen auf
`Vorhersage * -1 AS Vorhersage`
zu setzen. Leider blieben die Werte aber positiv. 
Kann es sein, dass der Server die php file cached? 
Edit: ja, scheint der cache zu sein. Jetzt geht's
Edit2: War mein Fehler, nicht der cache. Hatte das "*-1" noch an einer anderen Stelle. "-*-=+". Ich hätte die SQL-Abfrage genauer lesen sollen.

**Reply by wiggal** [reply]
Hab eingebaut in #146
